querydsl-jdo/src/main/java/com/querydsl/jdo/JDOQLSerializer.java

```diff
@@ -281,11 +282,17 @@ protected void visitOperation(Class<?> type, Operator operator, List<? extends E
             Constant<Class<?>> rightArg = (Constant<Class<?>>) args.get(1);
             append(rightArg.getConstant().getName());
 
-        } else if (operator == Ops.LIKE || operator == Ops.LIKE_ESCAPE) {
+        } else if (operator == Ops.LIKE || operator == Ops.LIKE_ESCAPE || operator == Ops.LIKE_IC || operator == Ops.LIKE_ESCAPE_IC) {
             @SuppressWarnings("unchecked") //This is the expected type for like
-            Expression<String> rightArg = (Expression<String>) args.get(1);
+            Expression<String> string = (Expression<String>) args.get(0);
+            @SuppressWarnings("unchecked") //This is the expected type for like
+            Expression<String> regex = ExpressionUtils.likeToRegex((Expression<String>) args.get(1), false);
+            if (operator == Ops.LIKE_IC || operator == Ops.LIKE_ESCAPE_IC) {
+                string = Expressions.stringOperation(Ops.LOWER, string);
```

Comment:
Locale.ENGLISH for now,and a general fix of locale handling in Querydsl could be the topic of another PR.